core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/AlterTableApplier.kt

```diff
@@ -30,6 +31,14 @@ internal fun LazyQuery.withAlterStatement(alter: SqlAlterTableStmt): LazyQuery {
       return@fold it.applyTo(lazyQuery)
     }
 
+    (alterTableRules as? MySqlAlterTableRules)?.alterTableAddIndex?.let {
```

Comment:
This feels gross because MySQL specific things are leaking in to the generic code. I could maybe move the rule to sql.bnf so it's not _only_ in MySQL.bnf, and then this would just work for things like Postgres as well?